server/eventlog/eventlog.go

```diff
@@ -23,41 +37,243 @@ func GetEventLogChunk(ctx context.Context, env environment.Env, req *elpb.GetEve
 		return nil, err
 	}
 
+	invocationInProgress := inv.InvocationStatus == int64(inpb.Invocation_PARTIAL_INVOCATION_STATUS)
 	c := chunkstore.New(env.GetBlobstore(), &chunkstore.ChunkstoreOptions{})
-	var intChunkId uint16
-	if len(req.ChunkId) > 0 {
-		n, err := strconv.ParseUint(req.ChunkId, 16, 16)
-		if err != nil {
+	eventLogPath := GetEventLogPathFromInvocationId(req.InvocationId)
+	startingIndex, _ := chunkstore.ChunkIndexAsUint16(inv.LastChunkId)
+
+	lastChunkIndex, err := c.GetLastChunkIndex(ctx, eventLogPath, startingIndex)
+	if err != nil {
+		if startingIndex == math.MaxUint16 {
+			// No chunks have been written for this invocation
+			if invocationInProgress {
+				// If the invocation is in progress, logs may be written in the future.
+				// Return an empty chunk with NextChunkId set to 0.
+				return &elpb.GetEventLogChunkResponse{
+					Chunk: &elpb.GetEventLogChunkResponse_Chunk{
+						Buffer: make([]byte, 0),
+					},
+					NextChunkId: chunkstore.ChunkIndexAsString(0),
+				}, nil
+			} else {
+				// If the invocation is not in progress, no logs will ever exist for
+				// this invocation. Return an empty chunk with NextChunkId left blank.
+				return &elpb.GetEventLogChunkResponse{
+					Chunk: &elpb.GetEventLogChunkResponse_Chunk{
+						Buffer: make([]byte, 0),
+					},
+				}, nil
+			}
+		} else {
+			// The last chunk id recorded in the invocation table is wrong. Most
```

Comment:
It's a bit unclear (at least to me) why `startingIndex != math.Uint16` implies last chunk id is wrong -- mind adding a bit more context here?

Reply:
np, done.